### DISCUSSION TITLE
[Question][Hw5] eval x

### DISCUSSION THREAD

**Discussion by ee12ha0220** [prosyslab-classroom/cs348-information-security#629] [Q&A]
In the semantics, eval<sub>x</sub><sup>#</sup>(m<sup>#</sup>) = m<sup>#</sup>(x). 

I thought that x was things like %a in the llvm IR, and we can find it by using is_argument in utils.ml. 
But I found out that is_argument could not find llvalues in the shape of %a. 

So  I want to ask
1) How does x looks like in llvm IR?
2) What values will make 'is_argument' true?

**Comment by RiceBiscuits** [comment]
1. please see this document (https://github.com/prosyslab-classroom/llvm-primer/blob/master/data-structure.md)

2. I don't know what you are actually doing, but please refer to the definition of the type argument. 
> This class represents an incoming formal argument to a Function.
> 
> A formal argument, since it is `‘formal’', does not contain an actual value but instead represents the type, argument number, and attributes of an argument for a specific function. When used in the body of said function, the argument of course represents the value of the actual argument that the function was called with.
(ref: https://llvm.org/doxygen/classllvm_1_1Argument.html)

**Comment by ee12ha0220** [comment]
So in implementing eval e mem, I want to know if e is x(in evalx#(m#) = m#(x)) or not. I thought using is_argument will verify this, but it doesn't. So how should I do this?

**Comment by RiceBiscuits** [comment]
Please read the document I mentioned in the previous comment.

**Comment by ee12ha0220** [comment]
I still can't quite get it, but I will read the materials again. Thank you for your time!
